haystack/components/converters/azure.py

```diff
@@ -137,7 +137,8 @@ def run(self, sources: List[Union[str, Path, ByteStream]], meta: Optional[List[D
             result = poller.result()
             azure_output.append(result.to_dict())
 
-            docs = self._convert_tables_and_text(result=result, meta=metadata)
+            merged_metadata = {**bytestream.meta, **metadata}
```

Comment:
I know this will not likely happen but whenever I see this type of code I'm concerned it will blow up with TypeError if either one of these is None for whatever reason. Wdyt we do:
```
merged_metadata = {**(bytestream.meta or {}), **(metadata or {})} 
```

Reply:
Not sure if it's worth it. `normalize_metadata` explicitly takes care about not returning `None` values and `ByteStream.meta` is not optional. So unless our typings are wrong I would rather rely on them instead of adding everywhere a `or {}` suffix, cluttering the code.

Reply:
Also we don't do it anywhere else. All the other converters merge it the same way.

Reply:
Ok, gotcha!